File: src/spec/resources/testprojects/javatest/src/main/java/com/example/petstore/models/PetPassport.java
package com.example.petstore.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import javax.annotation.Nullable;

public class PetPassport {

  @JsonProperty("pet")
  @Nullable
  public Pet pet;

  /** Base64-encoded primary thumbnail */
  @JsonProperty("thumbnail")
  @Nullable
  public byte[] thumbnail;

  /** Base64-encoded scans of each passport page */
  @JsonProperty("scans")
  @Nullable
  public List<byte[]> scans = new ArrayList<>();

  @JsonProperty("issuedAt")
  @Nullable
  public OffsetDateTime issuedAt;
}

File: src/spec/resources/testprojects/csharptest/src/PetstoreClient/Models/PetPassport.cs
#pragma warning disable CA1002 // Do not expose generic lists
#pragma warning disable CA1056 // URI-like properties should not be strings
#pragma warning disable CA1711 // Identifiers should not have incorrect suffix
#pragma warning disable CA1819 // Properties should not return arrays
#pragma warning disable CA2227 // Collection properties should be read only

using System.Text.Json.Serialization;

namespace PetstoreClient.Models;

public class PetPassport
{
    [JsonPropertyName("pet")]
    public Pet? Pet { get; set; }

    /// <summary>
    /// Base64-encoded primary thumbnail
    /// </summary>
    [JsonPropertyName("thumbnail")]
    public byte[]? Thumbnail { get; set; }

    /// <summary>
    /// Base64-encoded scans of each passport page
    /// </summary>
    [JsonPropertyName("scans")]
    public List<byte[]>? Scans { get; set; }

    [JsonPropertyName("issuedAt")]
    public DateTimeOffset? IssuedAt { get; set; }
}
